tests/Events/ProgressToClinicalEventTest.cpp

```
@@ -0,0 +1,95 @@
+#include "Utils/Cli.h"
+#include "fixtures/TestFileGenerators.h"
+#include "Treatment/ITreatmentCoverageModel.h"
+#include "Utils/Random.h"
+
+using namespace ::testing;
+
+// Minimal deterministic Random stub that returns a fixed value for random_flat
+class TestRandomFixed : public utils::Random {
+public:
+    explicit TestRandomFixed(double fixed_value) : utils::Random(nullptr, 42), fixed_(fixed_value) {}
+    double random_flat(double from, double to) override { (void)from; (void)to; return fixed_; }
```

Clang (14.0.6)

[error] 2 adjacent parameters of 'random_flat' of similar type ('double') are easily swapped by mistake

(bugprone-easily-swappable-parameters,-warnings-as-errors)